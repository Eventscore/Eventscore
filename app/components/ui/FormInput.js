//<FormInput></FormInput>
import React, { Component, PropTypes } from 'react';
import { FormInput } from 'react-native-elements';

// Consts and Libs
import { AppColors, AppFonts } from '@theme/';

class CustomFormInput extends Component {
  static propTypes = {
    containerStyle: PropTypes.oneOfType([
      PropTypes.array,
      PropTypes.shape({}),
    ]),
    inputStyle: PropTypes.oneOfType([
      PropTypes.array,
      PropTypes.shape({}),
    ]),
  }

  static defaultProps = {
    containerStyle: [],
    inputStyle: [],
  }

  inputProps = () => {
    // Defaults
    const props = {
      ...this.props,
      containerStyle: [{
        borderBottomColor: AppColors.border,
        borderBottomWidth: 1,
        backgroundColor: 'rgba(255,255,255,0.05)',
        marginTop: 10,
        marginLeft: 20,
        marginRight: 20,
      }],
      inputStyle: [{
        color: AppColors.textPrimary,
        fontFamily: AppFonts.base.family,
        paddingHorizontal: 10,
        paddingVertical: 3,
      }],
    };

    if (this.props.containerStyle) {
      props.containerStyle.push(this.props.containerStyle);
    }

    if (this.props.inputStyle) {
      props.inputStyle.push(this.props.inputStyle);
    }

    return props;
  }

  render = () => <FormInput {...this.inputProps()} />
}

export default CustomFormInput;
