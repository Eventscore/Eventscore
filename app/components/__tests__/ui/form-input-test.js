/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import FormInput from '@ui/FormInput';

it('FormInput renders correctly', () => {
  const tree = renderer.create(
    <FormInput value={'John Smith'} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
