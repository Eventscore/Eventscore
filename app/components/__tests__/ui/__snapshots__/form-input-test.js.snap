// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#bdc6cf",
        "borderBottomWidth": 1,
        "marginLeft": 20,
        "marginRight": 20,
      },
      Array [
        Object {
          "backgroundColor": "rgba(255,255,255,0.05)",
          "borderBottomColor": "#D0D1D5",
          "borderBottomWidth": 1,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 10,
        },
        Array [],
      ],
    ]
  }
>
  <TextInput
    autoCapitalize={undefined}
    autoCorrect={undefined}
    autoFocus={undefined}
    blurOnSubmit={undefined}
    clearButtonMode={undefined}
    clearTextOnFocus={undefined}
    dataDetectorTypes={undefined}
    defaultValue={undefined}
    editable={undefined}
    enablesReturnKeyAutomatically={undefined}
    inlineImageLeft={undefined}
    inlineImagePadding={undefined}
    keyboardAppearance={undefined}
    keyboardType={undefined}
    maxLength={undefined}
    multiline={undefined}
    numberOfLines={undefined}
    onBlur={undefined}
    onChange={undefined}
    onChangeText={undefined}
    onContentSizeChange={undefined}
    onEndEditing={undefined}
    onFocus={undefined}
    onKeyPress={undefined}
    onLayout={undefined}
    onSelectionChange={undefined}
    onSubmitEditing={undefined}
    placeholder={undefined}
    placeholderTextColor={undefined}
    returnKeyLabel={undefined}
    returnKeyType={undefined}
    secureTextEntry={undefined}
    selectTextOnFocus={undefined}
    selectionColor="#86939e"
    selectionState={undefined}
    style={
      Array [
        Object {
          "color": "#86939e",
          "fontSize": 17.5,
          "height": 36,
        },
        Array [
          Object {
            "color": "#222222",
            "fontFamily": "HelveticaNeue",
            "paddingHorizontal": 10,
            "paddingVertical": 3,
          },
          Array [],
        ],
      ]
    }
    underlineColorAndroid={undefined}
    value="John Smith"
  />
</View>
`;
